### PR TITLE
feat: validate locales compatibility [PHX-2921]

### PR DESCRIPTION
**Pull request by Vinz93**
## Summary 

- Adds Locales validation
- Run Locales validation within validate-changeset-task
- When management token fails it throws an Authorization error

### CODE REVIEW

File: src/engine/client/index.ts [hunk 1]
```diff
@@ -139,6 +139,13 @@ export const createClient = ({
           })
         },
       },
+      locales: {
```
Comment (marcolink) [inline src/engine/client/index.ts:142]: Could we also use the CDA to fetch locales?
https://www.contentful.com/developers/docs/references/content-delivery-api/#/reference/locales/locale-collection/get-all-locales-of-a-space/console/js

File: src/engine/validations/validate-locales.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const sourceEnvironmentId = context.changeset.sys.source.sys.id
+  let sourceLocales, targetLocales
+
+  try {
+    ;[sourceLocales, targetLocales] = await Promise.all([
+      context.client.cma.locales.getMany({ environment: sourceEnvironmentId }),
+      context.client.cma.locales.getMany({ environment: context.environmentId }),
+    ])
+  } catch (err) {
+    throw new AuthorizationErrorForApply(context)
```
Comment (marcolink) [inline src/engine/validations/validate-locales.ts:14]: This assumes that the only error possible is a 401, is that what we want?

File: src/engine/errors.ts [hunk 3]
```diff
@@ -14,6 +14,14 @@ export interface LimitsExceededContext {
   affectedEntities: AffectedEntities
 }
 
+export class AuthorizationErrorForApply extends ContentfulError {
```
Comment (marcolink) [inline src/engine/errors.ts:17]: Do we need a dedicated authorization error for the apply command, or could both commands share the same error?

File: src/engine/client/index.ts (hunk 1 above)
Comment (Vinz93) [reply src/engine/client/index.ts:142]: That would mean that we need to ask for CDA token for the apply command, wouldn't it? I am not sure if we want that

Comment (marcolink) [reply src/engine/client/index.ts:142]: Good point @Vinz93

File: src/engine/errors.ts (hunk 3 above)
Comment (Vinz93) [reply src/engine/errors.ts:17]: We are not throwing this error in create-changeset yet, I would say as soon as we start to use it in both place we can adjust make it more generic, but atm the context that expects and the message is custom only for apply and cma token

File: src/engine/validations/validate-locales.ts (hunk 2 above)
Comment (Vinz93) [reply src/engine/validations/validate-locales.ts]: adjusted 👍

File: test/e2e/commands/index.test.ts [hunk 4]
```diff
@@ -77,7 +77,9 @@ describe('Command flow - create and apply', () => {
       cmaToken: 'invalid-token',
     }))
     .it('should fail to apply changes with invalid token', async (ctx) => {
-      expect(ctx.stdout).to.contain('Error: An error occurred while deleting an entry.')
+      expect(ctx.stdout).to.contain(
```
Comment (marcolink) [inline test/e2e/commands/index.test.ts:80]: ❤️ This is a way better DX 👏

**Review by marcolink — APPROVED** [review]

**Pull request merged**